rust/datafusion/src/execution/physical_plan/planner.rs

```diff
@@ -306,32 +305,25 @@ impl DefaultPhysicalPlanner {
                 input_schema,
                 data_type.clone(),
             ),
-            Expr::ScalarFunction {
-                name,
-                args,
-                return_type,
-            } => match ctx_state.scalar_functions.get(name) {
-                Some(f) => {
-                    let mut physical_args = vec![];
-                    for e in args {
-                        physical_args.push(self.create_physical_expr(
-                            e,
-                            input_schema,
-                            ctx_state.clone(),
-                        )?);
+            Expr::ScalarFunction { name, args, .. } => {
+                match ctx_state.scalar_functions.get(name) {
+                    Some(f) => {
+                        let mut physical_args = vec![];
+                        for e in args {
+                            physical_args.push(self.create_physical_expr(
+                                e,
+                                input_schema,
+                                ctx_state.clone(),
+                            )?);
+                        }
+                        function(f, physical_args, input_schema)
```

Comment:
Note how `function` is called here. This is responsible for coercing udf's arguments to valid signatures.